helsingborg-stad/app-mitt-helsingborg
# Fix/image uploader

## Explain the changes you’ve made

Fix the imageUploader component, so that it is in usable shape and adheres to the Figma design. 

## Explain why these changes are made

We need it for stickprov. 

## Explain your solution

Basically rewrote the component from scratch, keeping just some basics from the earlier implementation. 
One big improvement that I made today after the group discussion was to forgo the use of base64-encoded strings, and instead fetch blobs from disk directly, and upload those. Then you do not have to deal with different encoding standards and it feels generally much safer. 

Otherwise I've just followed the updated Figma design. Still need some updated design for the remove image button, but here we are waiting for design, and the current implementation is good enough for a first release (according to Maria, haven't checked with Gustav). 

The only major thing missing in the implementation is deleting files from the cloud. This has to be implemented in the API (and then in the app), so I just have a placeholder here for now. 


## How to test the changes?

Checkout the branch, run yarn to install a required new library for handling images. Then you can go into the storybook and check the Image Uploader story, or you can go in to the Test form on develop, at the end of the first page there's an image uploader. Test that the images persists if you go back and forth between steps. 

Oh, every time you select an image, it's uploaded to the S3 bucket. I don't think this will be expensive unless someone goes completely crazy with it, but just be aware...


## Screenshots (optional)
![imageuploader2](https://user-images.githubusercontent.com/7421890/104603894-29f8bd00-567d-11eb-855e-e1bec7b45b22.png)
![imageuploader1](https://user-images.githubusercontent.com/7421890/104603896-2b29ea00-567d-11eb-828c-699e06ce60ab.png)
